src/LodeRunner.Core/Models/TestRunPayload.cs

```diff
@@ -0,0 +1,111 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Runtime.CompilerServices;
+using LodeRunner.Core.SchemaFilters;
+using Swashbuckle.AspNetCore.Annotations;
+
+namespace LodeRunner.Core.Models
+{
+    /// <summary>
+    /// TestRun Payload.
+    /// </summary>
+    [SwaggerSchemaFilter(typeof(TestRunPayloadSchemaFilter))]
+    public class TestRunPayload : BasePayload, INotifyPropertyChanged
+    {
+        // Composite TestRun object to hold data
+        private readonly TestRun testRun = new ();
+
+        /// <summary>
+        /// Gets or sets the name.
+        /// </summary>
+        /// <value>
+        /// The name.
+        /// </value>
+        public string Name { get => this.testRun.Name; set => this.SetField(value); }
+
+        /// <summary>
+        /// Gets or sets the LoadTestConfig.
+        /// </summary>
+        /// <value>
+        /// The LoadTestConfig.
+        /// </value>
+        public LoadTestConfig LoadTestConfig { get => this.testRun.LoadTestConfig; set => this.SetField(value); }
+
+        /// <summary>
+        /// Gets or sets the LoadClients.
+        /// </summary>
+        /// <value>
+        /// The LoadClients.
+        /// </value>
+        public List<LoadClient> LoadClients { get => this.testRun.LoadClients; set => this.SetField(value); }
+
+        /// <summary>
+        /// Gets or sets the CreatedTime.
+        /// </summary>
+        /// <value>
+        /// The CreatedTime.
+        /// </value>
+        public DateTime CreatedTime { get => this.testRun.CreatedTime; set => this.SetField(value); }
+
+        /// <summary>
+        /// Gets or sets the StartTime.
+        /// </summary>
+        /// <value>
+        /// The StartTime.
+        /// </value>
+        public DateTime StartTime { get => this.testRun.StartTime; set => this.SetField(value); }
+
+        /// <summary>
+        /// Sets mock TestRun payload data.
+        /// </summary>
+        /// <param name="name">TestRun name.</param>
+        public void SetMockData(string name)
```

Comment:
Let's move the mock creation out of the domain model proper.  Use a factory method in the test object or create an extension method in the test context to act as a factory method.

Reply:
done! Thanks @kforeverisback